syntax = "proto2";
option cc_generic_services = true;
package blp;

message SyncRequest {
    required string slave_id = 1;
    optional string last_message_id = 2;
}

message DataItem {
    required string message_id = 1;
    required string key = 2;
    required string value = 3;
}

message SyncResponse {
    repeated DataItem items = 1;
    optional bool is_full_sync = 2;
    optional bool has_more = 3;
}

service SyncService {
    rpc SyncData(SyncRequest) returns (SyncResponse);
}
